airflow/providers/google/cloud/operators/workflows.py

```diff
@@ -0,0 +1,638 @@
+        self.log.info("Canceling execution %s", self.execution_id)
+        execution = hook.cancel_execution(
+            workflow_id=self.workflow_id,
+            execution_id=self.execution_id,
+            location=self.location,
+            project_id=self.project_id,
+            retry=self.retry,
+            timeout=self.timeout,
+            metadata=self.metadata,
+        )
+        return Execution.to_dict(execution)
+
+
+class WorkflowExecutionsListExecutionsOperator(BaseOperator):
+    """
+    Returns a list of executions which belong to the
+    workflow with the given name. The method returns
+    executions of all workflow revisions. Returned
+    executions are ordered by their start time (newest
+    first).
+
+    :param workflow_id: Required. The ID of the workflow to be created.
+    :type workflow_id: str
+    :param project_id: Required. The ID of the Google Cloud project the cluster belongs to.
+    :type project_id: str
+    :param location: Required. The Cloud Dataproc region in which to handle the request.
+    :type location: str
+    :param retry: A retry object used to retry requests. If ``None`` is specified, requests will not be
+        retried.
+    :type retry: google.api_core.retry.Retry
+    :param timeout: The amount of time, in seconds, to wait for the request to complete. Note that if
+        ``retry`` is specified, the timeout applies to each individual attempt.
+    :type timeout: float
+    :param metadata: Additional metadata that is provided to the method.
+    :type metadata: Sequence[Tuple[str, str]]
+    """
+
+    template_fields = ("location", "workflow_id")
+
+    def __init__(
+        self,
+        *,
+        workflow_id: str,
+        location: str,
+        project_id: str,
+        retry: Optional[Retry] = None,
+        timeout: Optional[float] = None,
+        metadata: Optional[Sequence[Tuple[str, str]]] = None,
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        self.workflow_id = workflow_id
+        self.location = location
+        self.project_id = project_id
+        self.retry = retry
+        self.timeout = timeout
+        self.metadata = metadata
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+
+    def execute(self, context):
+        hook = WorkflowsHook(gcp_conn_id=self.gcp_conn_id, impersonation_chain=self.impersonation_chain)
+        self.log.info("Retrieving executions for workflow %s", self.workflow_id)
+        execution_iter = hook.list_executions(
+            workflow_id=self.workflow_id,
+            location=self.location,
+            project_id=self.project_id,
+            retry=self.retry,
+            timeout=self.timeout,
+            metadata=self.metadata,
+        )
+        return [Execution.to_dict(e) for e in execution_iter]
```

Comment:
The number of Executions returned can get pretty big (millions), so let's limit to executions started in the past 60 minutes. The LIST API doesn't support filtering today, so you'll need to do it yourself. You can use Execution.startTime for it:
https://cloud.google.com/workflows/docs/reference/executions/rest/v1beta/projects.locations.workflows.executions#State

Reply:
I will make it configurable then 👍

Reply:
Done